urlaf656038a0_Salihan04_MOBLIMA_tgz-pJ8-com_moblima_util_dataTestJ8/src/com/moblima/presentation/MovieListing_CLI.java

```diff
@@ -0,0 +1,457 @@
+package com.moblima.presentation;
+
+import com.moblima.businesslogic.MovieBL;
+import com.moblima.model.Movie;
+import com.moblima.model.MovieRating;
+import com.moblima.model.MovieStatus;
+import com.moblima.model.MovieType;
+import com.moblima.util.ConsoleReader;
+
+import java.util.Scanner;
+
+public class MovieListing_CLI {
+	private MovieBL movieBL;
+	
+	public MovieListing_CLI(){
+		movieBL = new MovieBL();
+	}
+	
+	public void createMovie(){
+		int choice;
+		String title,details,language, cast, director, opening, runtime;
+		MovieRating rating=null;
+		MovieType type=null;
+		MovieStatus status=null;
+		char isBlockbuster;
+		boolean blockBuster;
+		
+		System.out.println();
+		System.out.println("Create Movie");
+		System.out.println("------------");
+		System.out.print("Enter the Movie Title : ");
+		title = ConsoleReader.readString();
+		System.out.println();
+		
+		System.out.println("Movie Type");
+		System.out.println("----------");
+		System.out.println("1) DIGITAL");
+		System.out.println("2) 3D");
+		System.out.print("Choose the Movie Type: ");
+		choice = ConsoleReader.readIntInput();
+		if(choice>2 || choice<1){
+			System.out.println("Error invalid choice!");
+			return;
+		}
+		System.out.println();
+		
+		switch(choice){
+			case 1:
+				type = MovieType.DIGITAL;
+				break;
+				
+			case 2:
+				type = MovieType._3D;
+				break;
+		}
+		
+		System.out.println("Is this a Blockbuster movie? (Y/N)");
+		isBlockbuster = ConsoleReader.readChar();
+		if(isBlockbuster == 'Y')
+			blockBuster = true;
+		else if(isBlockbuster == 'N')
+			blockBuster = false;
+		else{
+			System.out.println("Invalid Option!");
+			return;
+		}
+		
+		System.out.println("Movie Rating");
+		System.out.println("------------");
+		System.out.println("1) G");
+		System.out.println("2) PG");
+		System.out.println("3) PG13");
+		System.out.println("4) NC16");
+		System.out.println("5) M18");
+		System.out.println("6) R21");
+		System.out.print("Choose the Movie Rating: ");
+		choice = ConsoleReader.readIntInput();
+		if(choice>6 || choice<1){
+			System.out.println("Invalid Choice!");
+			return;
+		}
+		System.out.println();
+		
+		switch(choice){
+			case 1:
+				rating = MovieRating.G;
+				break;
+			
+			case 2:
+				rating = MovieRating.PG;
+				break;
+			
+			case 3:
+				rating = MovieRating.PG13;
+				break;
+			
+			case 4:
+				rating = MovieRating.NC16;
+				break;
+			
+			case 5:
+				rating = MovieRating.M18;
+				break;
+			
+			case 6:
+				rating = MovieRating.R21;
+				break;
+		}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 84:108, 283:307.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.